lib/easybill-rest-client/generic_api.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+module EasybillRestClient
+  module GenericApi
+    def self.new(resource_name, resource_class)
```

Comment:
I described my reasons for doing it like this. You weigh the decision. I don't find the current approach too complex(ever seen `Struct.new(:foo, :bar)` 😃 ?), but it's definitely less straightforward than just typing everything explicitly.